tool/world_options.re

```diff
@@ -0,0 +1,300 @@
+            onclick: fn(input: string) {
+                match input.parse_float() {
+                    Result::Ok(dilation) => {
+                        SETTINGS.time_dilation = dilation;
+                        SETTINGS.store();
+                        Tas::set_time_dilation(dilation);
+                    },
+                    Result::Err(e) => {},
+                }
+            },
+            onchange: fn(input: string) {},
+        }),
+        UiElement::FloatInput(FloatInput {
+            label: Text { text: "Gravity" },
+            input: f"{SETTINGS.gravity}",
+            onclick: fn(input: string) {},
+            onchange: fn(input: string) {
+                match input.parse_float() {
+                    Result::Ok(gravity) => {
+                        Tas::set_gravity(gravity);
+                        SETTINGS.gravity = gravity;
+                        SETTINGS.store();
+                    },
+                    Result::Err(e) => {},
+                }
+            },
+        }),
+        UiElement::FloatInput(FloatInput {
+            label: Text { text: "Kill Z" },
+            input: f"{SETTINGS.kill_z}",
+            onclick: fn(input: string) {
+                match input.parse_float() {
+                    Result::Ok(kill_z) => {
+                        Tas::set_kill_z(kill_z);
+                        SETTINGS.kill_z = kill_z;
+                        SETTINGS.store();
+                    },
+                    Result::Err(e) => {},
+                }
+            },
+            onchange: fn(input: string) {},
+        }),
+        UiElement::FloatInput(FloatInput {
+            label: Text { text: "Outro Dilated Duration" },
+            input: f"{SETTINGS.outro_dilated_duration}",
+            onclick: fn(input: string) {},
+            onchange: fn(input: string) {
+                match input.parse_float() {
+                    Result::Ok(duration) => {
+                        Tas::set_outro_dilated_duration(duration);
+                        SETTINGS.outro_dilated_duration = duration;
+                        SETTINGS.store();
+                    },
+                    Result::Err(e) => {},
+                }
+            },
+        }),
+        UiElement::FloatInput(FloatInput {
+            label: Text { text: "Outro Time Dilation" },
+            input: f"{SETTINGS.outro_time_dilation}",
+            onclick: fn(input: string) {},
+            onchange: fn(input: string) {
+                match input.parse_float() {
+                    Result::Ok(dilation) => {
+                        Tas::set_outro_time_dilation(dilation);
+                        SETTINGS.outro_time_dilation = dilation;
+                        SETTINGS.store();
+                    },
+                    Result::Err(e) => {},
+                }
+            },
+        }),
+        UiElement::FloatInput(FloatInput {
+            label: Text { text: "Time Of Day" },
```

Comment:
Should this be a setting persisted over restarts, which can be toggled on and off?

Reply:
Not sure about the toggle part, but yeah, the time of day can definitely be stored in settings and restored.